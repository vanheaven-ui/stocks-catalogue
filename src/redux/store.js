import { compose, createStore } from 'redux';
import rootReducer from './reducers';

const storeEnhancers = compose(
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(), // eslint-disable-line
);

const store = createStore(rootReducer, storeEnhancers);

export default store;
