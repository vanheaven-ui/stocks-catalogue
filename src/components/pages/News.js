import { Badge } from 'react-bootstrap';
import { useDispatch } from 'react-redux';
import { getNews } from '../../redux/actions';

const News = () => {
  const dispatch = useDispatch();
  // const { data: news } =
  // useFetch(`${process.env.REACT_APP_NEWS_API_KEY}apikey=
  // ${process.env.REACT_APP_STOCKS_API_KEY}`);
  // localStorage.setItem('news', JSON.stringify(news));
  const news = JSON.parse(localStorage.getItem('news'));
  dispatch(getNews(news));
  return (
    <section className="news">
      {news && news.map(item => (
        <article className="news-article" key={item.symbol}>
          <header>
            { item.title && (
              <h2>
                {item.title}
                <Badge variant="danger">{item.symbol}</Badge>
              </h2>
            )}
            <div className="time-author">
              {item.publishedDate && <small>{item.publishedDate}</small>}
              <div className="author">
                {item.site && <h4>{item.site}</h4>}
                {item.url && <a href={item.url}>Website</a>}
              </div>
              <div className="news-img">
                {item.image && <img src={item.image} alt="news-hero" /> }
              </div>
              <div className="body">
                {item.text && <p>{item.text}</p>}
              </div>
            </div>
          </header>
        </article>
      ))}
    </section>
  );
};

export default News;
