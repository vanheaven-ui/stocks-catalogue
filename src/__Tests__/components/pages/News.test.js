import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render, cleanup } from '@testing-library/react'; 
import { Provider } from 'react-redux';
import News from '../../../components/pages/News';
import rootNode from '../../helpers';
import store from '../../../redux/store';
import '@testing-library/jest-dom';

describe('News page', () => {
  afterEach(cleanup);

  test('renders without crashing', () => {
    ReactDOM.render(<Provider store={store}><News /></Provider>, rootNode());
  });

  test('doesnot change unexpectedly', () => {
    const NotFoundTree = renderer.create(
      <Provider store={store}><News /></Provider>,
    ).toJSON();

    expect(NotFoundTree).toMatchSnapshot();
  });

  test('renders the correct wrapper element', () => {
    const { container } = render(<Provider store={store}><News /></Provider>);
    expect(container.firstChild.classList.contains('news')).toBe(true);
  });
});
