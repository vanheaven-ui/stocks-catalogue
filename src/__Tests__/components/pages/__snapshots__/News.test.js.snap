// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound page doesnot change unexpectedly 1`] = `
<section
  className="news"
>
  <div
    className="container"
    style={
      Object {
        "backgroundColor": "#fff",
        "textAlign": "center",
      }
    }
  />
</section>
`;
